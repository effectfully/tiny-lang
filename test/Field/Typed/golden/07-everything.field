ext eb : bool;
ext ef : field;
ext ev : vector;



let a : bool = T;
let b : bool = F;

let or'  : bool = a or b;
let or'' : bool = a or eb;
let and' : bool = or'' and b;
let xor' : bool = and' xor or'';



let f : field = 0;
let g : field = 1;
let h : field = 2;

let feq : bool = g  == f;
let fe' : bool = ef == f;
let fle : bool = h  <= f;
let flt : bool = f  <  g;
let fge : bool = g  >= h;
let fgt : bool = h  >  f;

let add : field = f + g;
let ad' : field = f + ef;
let sub : field = g - h;
let mul : field = h * f;
let div : field = f / g;



let v : vector = {T, F, T};
let bat : bool =  v[f];
let ba' : bool = ev[f];

let not'  : bool  = not a;
let neq0' : bool  = neq0 f;

let neg' : field = neg g;
let inv' : field = inv h;

let unp : vector = unpack h;



let let' : bool = T;
assert T;
for j = 0 to 2 do end;
let if' : bool = if T then T else T;


let asf : field  = 1 : field;
let asb : bool   = T : bool;
let asv : vector = { T } : vector;
