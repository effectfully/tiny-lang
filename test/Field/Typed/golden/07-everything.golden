Program [?i_36,i_37,#i_38] Statements [ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?a_39}) (EConst (UniConst Bool True)),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?b_40}) (EConst (UniConst Bool False)),ELet (UniVar {_uniVarUni = Vector, _uniVarVar = #v_41}) (EConst (UniConst Vector [True,False,True])),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?e_42}) (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?i_36})),ELet (UniVar {_uniVarUni = Field, _uniVarVar = e_43}) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = i_37})),ELet (UniVar {_uniVarUni = Vector, _uniVarVar = #e_44}) (EVar (UniVar {_uniVarUni = Vector, _uniVarVar = #i_38})),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?or_45}) (EAppBinOp Or (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?a_39})) (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?b_40}))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?and_46}) (EAppBinOp Or (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?or_45})) (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?b_40}))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?xor_47}) (EAppBinOp Or (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?and_46})) (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?or_45}))),ELet (UniVar {_uniVarUni = Field, _uniVarVar = f_48}) (EConst 0),ELet (UniVar {_uniVarUni = Field, _uniVarVar = g_49}) (EConst 1),ELet (UniVar {_uniVarUni = Field, _uniVarVar = h_50}) (EConst 2),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?feq_51}) (EAppBinOp FEq (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_49})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_48}))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?fle_52}) (EAppBinOp FLe (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_50})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_48}))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?flt_53}) (EAppBinOp FLt (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_48})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_49}))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?fge_54}) (EAppBinOp FGe (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_49})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_50}))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?fgt_55}) (EAppBinOp FGt (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_50})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_48}))),ELet (UniVar {_uniVarUni = Field, _uniVarVar = add_56}) (EAppBinOp Add (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_48})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_49}))),ELet (UniVar {_uniVarUni = Field, _uniVarVar = sub_57}) (EAppBinOp Sub (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_49})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_50}))),ELet (UniVar {_uniVarUni = Field, _uniVarVar = mul_58}) (EAppBinOp Mul (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_50})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_48}))),ELet (UniVar {_uniVarUni = Field, _uniVarVar = div_59}) (EAppBinOp Div (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_48})) (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_49}))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?bat_60}) (EAppBinOp BAt (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_48})) (EVar (UniVar {_uniVarUni = Vector, _uniVarVar = #v_41}))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?not_61}) (EAppUnOp Not (EVar (UniVar {_uniVarUni = Bool, _uniVarVar = ?a_39}))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?neq0_62}) (EAppUnOp Neq0 (EVar (UniVar {_uniVarUni = Field, _uniVarVar = f_48}))),ELet (UniVar {_uniVarUni = Field, _uniVarVar = neg'_63}) (EAppUnOp Neg (EVar (UniVar {_uniVarUni = Field, _uniVarVar = g_49}))),ELet (UniVar {_uniVarUni = Field, _uniVarVar = inv'_64}) (EAppUnOp Inv (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_50}))),ELet (UniVar {_uniVarUni = Vector, _uniVarVar = #unpack_65}) (EAppUnOp Unp (EVar (UniVar {_uniVarUni = Field, _uniVarVar = h_50}))),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?let_66}) (EConst (UniConst Bool True)),EAssert (EConst (UniConst Bool True)),ELet (UniVar {_uniVarUni = Field, _uniVarVar = j_67}) (EConst 0),ELet (UniVar {_uniVarUni = Field, _uniVarVar = j_68}) (EConst 1),ELet (UniVar {_uniVarUni = Field, _uniVarVar = j_69}) (EConst 2),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?if_70}) (EIf (EConst (UniConst Bool True)) (EConst (UniConst Bool True)) (EConst (UniConst Bool True))),ELet (UniVar {_uniVarUni = Field, _uniVarVar = asf_71}) (EConst 1),ELet (UniVar {_uniVarUni = Bool, _uniVarVar = ?as_72}) (EConst (UniConst Bool True)),ELet (UniVar {_uniVarUni = Vector, _uniVarVar = #as_73}) (EConst (UniConst Vector [True]))]